docs/faq.rst

```diff
@@ -391,6 +391,17 @@ How can I configure and debug the exact version exported by my package?
 
     (This requires the ``setuptools-scm`` package is installed in your environment)
 
+Why do I see the error ``AssertionError: cant parse version ...``?
+    There are some limitations with ``setuptools-scm`` and in some edge cases
+    it might fail to parse specific tags.
+    If that is happening to your project (other than removing the problematic tag),
```

Comment:
I am not super sure about it, but should it rather say that `setuptools-scm` expects version tags according to the [semantic versioning scheme](https://semver.org/). Thus, rename your tags to look like `v1.3` or `v1.3.2`.

Reply:
Currently `setuptools-scm` seems to aim for ignoring tags that don't resemble a version tag.
For example:
- it would ignore a tag named `draft`
- it would parse a tag named `draft-1` and give back version `1` (it assumes that the tag is in the format `<package name>-<version number>`)

However there is an edge case which results in an exception when no `-` is used (e.g. `draft1`).

I noticed that `setuptools-scm` has an [issue open](https://github.com/pypa/setuptools_scm/issues/449) which seems to indicate that the problem is fixed. So I asked the developers about this behaviour in the discord channel: it might just be a limitation that `setuptools-scm` chooses to not handle, rather than an error.

What is weird to me is that the code path checks twice the tag format. The first time `setuptools-scm` sees the tag, it complains with a warning but keeps going on, the second time it raises an error via an assert statement.

That looks a bit odd. If the intention is to not support this scenario, it would seem better to just fail the first time it checks. Moreover, assert statements in production code are (most of the times) added when the developer already believes that the assertion will never fail and that all the edge cases have already been handled. But yeah, I don't know much about `setuptools-scm` code base so let's see if they can clarify that detail...

Reply:
Ah! Thanks for the explanation. So from a user's perspective, one could say:

> The way ``setuptools-scm`` parses git tags has some edge cases leading to this assertion error. The best way is to either remove the tag or rename it to either a proper semantic version like `v0.4.2` or a word without a number like `draft`.